src/library/Flex/Flex.js

```diff
@@ -63,60 +66,123 @@ export type Values = boolean | null | number | string;
 const getAlignment = (value: string): string =>
   ['start', 'end'].indexOf(value) !== -1 ? `flex-${value}` : value;
 
-const getGutterSize = (theme: Object, value: number | string): string =>
-  typeof value === 'number'
-    ? pxToEm(value / 2)
-    : `${parseFloat(theme[`space_inline_${value}`] || value) / 2}em`;
-
-const getMargin = (
-  gutter: string,
+const getGutterSize = (
   theme: Object,
   value: number | string
-): string =>
-  value ? `calc(${getMeasurement(theme, value)} - ${gutter})` : `-${gutter}`;
+): number | string =>
+  typeof value === 'number' && value !== 0
+    ? pxToEm(value)
+    : theme[`space_inline_${value}`] || value;
 
-const getMeasurement = (theme: Object, value: number | string): string => {
-  return (
-    theme[`space_inline_${value}`] ||
-    (typeof value === 'number' ? `${value}px` : value)
-  );
-};
+const getIndexedValue = (property, index) =>
+  property && Array.isArray(property) && index !== undefined
+    ? property[index]
+    : property;
 
 const getJustification = (value: string): string =>
   ['around', 'between', 'evenly'].indexOf(value) !== -1
     ? `space-${value}`
     : getAlignment(value);
 
+const getMarginOrGutter = ({
+  gutterWidth,
+  index,
+  margin,
+  marginEnd,
+  marginHorizontal,
+  marginStart,
+  start,
+  theme
+}) => {
+  return (
+    getIndexedValue(start ? marginStart : marginEnd, index) ||
+    getIndexedValue(marginHorizontal, index) ||
+    getIndexedValue(margin, index) ||
+    getGutterSize(theme, gutterWidth)
```

Comment:
I'm a little confused about the logic here; if there is a marginStart or marginEnd (& index), then there is never a marginHorizontal? It might be helpful to discuss in person.

Reply:
It's an order of precedence thing. If a user provides a marginStart/End, marginHorizontal, or margin prop, the top line (in the code above) wins over the next and over the next until it falls back to applying the gutter.